### DISCUSSION TITLE
Issue with building VS2017 Projects - Error MSB4041

### DISCUSSION THREAD

**Discussion by ravensorb** [cake-build/cake#3265] [Q&A] [answered]
I recently started porting over some of my projects to VS2017 and updating the csproj files using the "dotnet migrate" command.  Everything was going fine in terms of building (VS2017 and command line via dotnet build) until I got around to testing my cakebuild scripts.  I double checked everything and confirmed I was using the DotNetBuild alias and still no luck.   

The error I am getting is
```
Collections.csproj(1,1): error MSB4041: The default XML namespace of the project mu
t be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Proje
t> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.
```

After doing some digging and enabling Diagnostic level logging, I am getting the feeling that cakebuild is not following the new build guidelines?

```
Building Collections.sln
Executing: "C:/WINDOWS/Microsoft.NET/Framework64/v4.0.30319/MSBuild.exe" /m /v:normal /p:Configuration="Release" /p:TreatWarningsAsErrors=False /target:Build "E:/Projects/Collections.sln"
Microsoft (R) Build Engine version 4.7.2046.0
[Microsoft .NET Framework, version 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.
```

Any thoughts?  Is this a bug or am I missing something?

Update: Ok, I think I found it.  It looks like the bootstrapper was refusing to download a newer version of cake even after I deleted it.  

Suggestion:  Make build.ps1 autoupdating as well :)

**Comment by devlead** [comment, marked as answer]
DotNetCoreBuild is the alias your looking for 
http://cakebuild.net/api/Cake.Common.Tools.DotNetCore/DotNetCoreAliases/8E8365A3
